webextension/scripts/resource_list.js

```diff
@@ -16,14 +21,18 @@ function show_resource_data(url_name) {
       if (message.message === 'resource_list_show' && message.url === url_name) {
         vdata = message.data
         status = message.data.status
-        $('#current-status').text(status.charAt(0).toUpperCase() + status.slice(1))
+        $('#current-status').text(status_list[status])
+        if ('message' in vdata) {
+          $('#message').text(vdata.message)
+        }
         if ('resources' in vdata) {
           new_resource_length = vdata.resources.length
           vdata.resources.forEach((element) => {
             resource_list_data.add(element)
           })
         }
         if(new_resource_length>old_resource_length){
+          $('.loader').hide()
```

Comment:
The `status_list` and `#message` look really good to me.
Also, I would suggest hiding the loader inside `if (status === 'success')` instead of here.

Reply:
No, we also get the list of resource data sometimes when the status is pending !! So We have to show the loader here !

Reply:
Yes, I see that. I suggested it because the download is still not completed and removing the loader makes it feel as if it is completed. It is more like a choice, we can remove or keep the loader for this case. However, not hiding the loader will look like this, which may show the 'Processing'/pending/downloading state more clearly :-)

![Screenshot (117)](https://user-images.githubusercontent.com/42881874/86542471-a3c42700-bf33-11ea-98bc-a8dabda0b036.png)

Reply:
Oh, I got your point. Thank you ! I will add this in the next commit !